gateways/cybersource/request_builders.go

```diff
@@ -92,6 +114,62 @@ func buildAuthRequest(authRequest *sleet.AuthorizationRequest) (*Request, error)
 			},
 		},
 	}
+
+	// Apple Pay request
+	if authRequest.Cryptogram != "" {
+		request.PaymentInformation = &PaymentInformation{
+			TokenizedCard: &TokenizedCard{
+				Number: authRequest.CreditCard.Number,
+				ExpirationYear: strconv.Itoa(authRequest.CreditCard.ExpirationYear),
+				ExpirationMonth: fmt.Sprintf("%02d", authRequest.CreditCard.ExpirationMonth),
+				TransactionType: "1",	// in-app transaction
+				Cryptogram: authRequest.Cryptogram,
+			},
+		}
+
+		request.ProcessingInformation.PaymentSolution = "001"	// Apple Pay
+
+		switch authRequest.CreditCard.Network {
+		case sleet.CreditCardNetworkVisa:
+			request.PaymentInformation.TokenizedCard.Type = CardTypeVisa
+			request.ConsumerAuthenticationInformation = &ConsumerAuthenticationInformation{
+				Xid: authRequest.Cryptogram,
+				Cavv: authRequest.Cryptogram,
```

Comment:
Why do we want the cryptogram in both fields?

Reply:
From https://developer.cybersource.com/api/developer-guides/dita-payments/CreatingOnlineAuth/CreatingAuthReqPNT.html#id189DF0WI0HT_CreatingAuthReqPNT, it asks me to set both fields for Visa request.

Reply:
I can't see the docs but does it want Xid to be populated with the transaction ID rather than the cryptogram? The transaction ID is a separate piece of Apple Pay data

Reply:
Yeah, from the doc, it mentioned
```For Visa requests:
Set the commerceIndicator field under the processingInformation object to vbv or internet.
Set the xid field under the consumerAuthenticationInformation object to the 3D Secure cryptogram of the payment network token.
Set the cavv field under the consumerAuthenticationInformation object to the 3D Secure cryptogram of the payment network token.
```

And from Amex, it actually also use Xid for Cyrptogram when length is 40-byte. So I think Cybersource would also use Xid for Cryptogram for some use case?